DOCSP-51695: [IS Edits] Build a PDF Search Application with Vector Search and LLMs (#277)
mongodb/docs-atlas-architecture: source/solutions-library/pdf-search.txt

in PDF format. Manually finding and reviewing every piece of information is
time-consuming and can easily lead to expensive mistakes, such as
incorrect risk estimations.

`Retrieval-augmented generation
<https://www.mongodb.com/resources/basics/retrieval-augmented-generation>`__
(RAG) applications are a game changer for insurance companies, enabling
them to harness the power of unstructured data while promoting
accessibility and flexibility. They are especially helpful for PDFs, which are
common yet difficult to search. RAG makes PDF search more efficient and 
accurate. Now, users can simply type a question in natural language 
and the application will sift through the company data, provide an answer, 
summarize the content of the documents, and indicate the source of the 
information, including the page and paragraph where it was found.

In this `GitHub repo
<https://github.com/mongodb-industry-solutions/Insurance-PDF-Search>`__,
you will find detailed, step-by-step instructions on how to build a
PDF search application that combines MongoDB, Superduper, and LLMs. Our use
case for this solution focuses on a claims adjuster or an underwriter

> handling a specific case. Analyzing a PDF of guidelines that is associated
> with a specific customer helps determine the loss amount, in the event of an
> accident, or the new premium, in the case of a policy renewal. The application

Reviewer: I don't think we need these commas:

```suggestion
handling a specific case. Analyzing a PDF of guidelines that is associated
with a specific customer helps determine the loss amount in the event of an
accident, or the new premium in the case of a policy renewal. The application
```

answers questions and displays the relevant sections of
the document.

Combining `Atlas Vector Search
<https://www.mongodb.com/products/platform/atlas-vector-search>`__ and
`LLMs
<https://www.mongodb.com/resources/basics/large-language-models>`__ to
build RAG apps can directly impact the bottom line of an insurance
company. Visit the :ref:`Atlas Vector Search Quick Start <vector-search-quick-start>`

> page to try our semantic search tool now.

Reviewer: [Minor]

Consider writing this in the order "To (x), do (y)"

.. video:: https://youtu.be/_w9okXmzTJw

Building the Solution and Reference Architecture
------------------------------------------------

Combining MongoDB and Superduper allows you to build an information
retrieval system with ease. The process involves the following steps:

#. The user adds the PDFs that need to be searched.
    
#. A script scans the PDFs, creates the chunks, and vectorizes them (see

>    Figure 1). The script uses a sliding window methodology to produce
>    overlapping chunks. This ensures that you don't
>    lose important transitional data between chunks.

Reviewer: ```suggestion
   Figure 1). To ensure that you don't lose transitional data between chunks, the script uses a sliding window methodology to produce
   overlapping chunks.
```

framework for integrating AI models and workflows directly with and
across major databases for more flexible and scalable custom enterprise
AI solutions. It enables developers to build, deploy, and manage AI on
their existing infrastructure and data while using their preferred
tools, eliminating data migration and duplication.

**With Superduper.io, developers can:**

- Bring AI to their databases, which eliminates data pipelines and
  minimizes engineering efforts, time to production, and
  computation resources.

> - Implement AI workflows with any AI models and APIs, and on any type of

Reviewer: ```suggestion
- Implement AI workflows with any AI models and APIs on any type of
```

  data.

- Safeguard data by switching from APIs to hosting and fine-tuning their
  own models on their own infrastructure.

> - Easily switch from embedding models to other API providers. They can

Reviewer: ```suggestion
- Switch from embedding models to other API providers. They can
```

  also host their models on HuggingFace or other platforms.

Superduper.io provides an array of sample use cases and notebooks that
developers can use to get started, including vector search with MongoDB,
embedding generation, multimodal search, RAG, transfer learning, and
many more. This solution's demo is adapted from a
`previously developed app <https://github.com/SuperDuperDB/poc-volvo>`__ from
Superduper.io. 

Key Learnings
-------------

.. MAYA: this should go somewhere before key learnings.

- Build the solution following the instructions in this `Github repo
  <https://github.com/mongodb-industry-solutions/Insurance-PDF-Search/tree/main>`__.
  The solution is composed of two steps:

  - The initialization script breaks down the PDFs into chunks and then
    turns them into vector embeddings.

>   - The querying step allows the user to interrogate the documents.

Reviewer: Optional but maybe this should be an ordered list, if this is a list of steps (even if it's only 2)
